fix(deps): updated vulnerable dependencies

rootxrishabh opened the pull request:
**Purpose of PR?**:
This PR updates all the vulnerable dependencies alerted by the OSSF scorecard.
Fixes #
#1532 
**Does this PR introduce a breaking change?**
No
**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_


**Checklist:**
- [X] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

<!--

The PR title message must follow convention:
`<type>(<scope>): <subject>`.

Where: <br />
- `type` is to define what type of PR is this.
  Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script
    - `chore`     - changes not related to production code
    - `docs`      - changes related to documentation
    - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes
    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

- `scope` is a single word that best describes where the changes fit.
    - feature(`monitor`,`enforcer`)
    - test(`tests`, `bdd`)
    - chore(`build`)
- `subject` is a single line brief description of the changes made in the pull request.

-->

DelusionalOptimist committed:
send protected objects with state agent

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

DelusionalOptimist committed:
udpate protobufs

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

DelusionalOptimist committed:
refactor feeder and add reverse services

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

DelusionalOptimist committed:
iterate runtime socket map in order

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

DelusionalOptimist committed:
cleanup protos

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

DelusionalOptimist committed:
refactor: improve implementation by preserving interfaces

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

rootxrishabh committed:
fix(deps): updated vulnerable dependencies

Signed-off-by: rootxrishabh <risrock02@gmail.com>

rootxrishabh committed:
Merge branch 'kubearmor:main' into updateDependencies

rootxrishabh committed:
removed k8s.io replace directives

Signed-off-by: rootxrishabh <risrock02@gmail.com>

rootxrishabh committed:
Dependencies updated

Signed-off-by: rootxrishabh <risrock02@gmail.com>

rootxrishabh committed:
toolchain removed

Signed-off-by: rootxrishabh <risrock02@gmail.com>

rootxrishabh committed:
Go version changed to 1.20

Signed-off-by: rootxrishabh <risrock02@gmail.com>

standardized controller-runtime version

Signed-off-by: rootxrishabh <risrock02@gmail.com>

x/net updated

Signed-off-by: rootxrishabh <risrock02@gmail.com>

Tested snyk locally

Signed-off-by: rootxrishabh <risrock02@gmail.com>

test changes reverted

Signed-off-by: rootxrishabh <risrock02@gmail.com>

controller-runtime version reverted

Signed-off-by: rootxrishabh <risrock02@gmail.com>

version reverted to 1.20

Signed-off-by: rootxrishabh <risrock02@gmail.com>

test go.mod updated

Signed-off-by: rootxrishabh <risrock02@gmail.com>

test go.mod updated

Signed-off-by: rootxrishabh <risrock02@gmail.com>

Errors fixed

Signed-off-by: rootxrishabh <risrock02@gmail.com>

DelusionalOptimist commented on pkg/KubeArmorController/go.mod:
```diff
@@ -2,80 +2,72 @@ module github.com/kubearmor/KubeArmor/pkg/KubeArmorController
+	k8s.io/apiextensions-apiserver v0.29.0
+	k8s.io/apimachinery v0.29.0
+	k8s.io/client-go v0.29.0
+	sigs.k8s.io/controller-runtime v0.15.1
```
Is it possible to match at least the minor version of controller-runtime here with the Operator and KubeArmor. :eyes:

DelusionalOptimist reviewed (commented):
All other updates look good. :+1:

daemon1024 commented on protobuf/go.mod:
```diff
@@ -17,8 +17,8 @@ require (
-	golang.org/x/net v0.10.0 // indirect
-	golang.org/x/sys v0.8.0 // indirect
-	golang.org/x/text v0.9.0 // indirect
+	golang.org/x/net v0.17.0 // indirect
```
https://security.snyk.io/vuln/SNYK-GOLANG-GOLANGORGXCRYPTOSSH-6130669
```
golang.org/x/net@v0.17.0 › golang.org/x/crypto@v0.14.0 
```

Snyk failure cause

rootxrishabh replied:
Upgraded to golang.org/x/net@v0.20.0 uses [golang.org/x/crypto v0.18.0](https://github.com/golang/net/blob/v0.20.0/go.mod#L6), still snyk fails. Am I missing something?

rootxrishabh replied:
> Upgraded to golang.org/x/net@v0.20.0 uses [golang.org/x/crypto v0.18.0](https://github.com/golang/net/blob/v0.20.0/go.mod#L6), still snyk fails. Am I missing something?

Solved!
Could you provide a bit more insight on `security/snyk (nyrahul)` ?

Aryan-sharma11 replied:
https://security.snyk.io/vuln/SNYK-GOLANG-GITHUBCOMGOJOSEGOJOSE-6070736
Snyk failure cause

rootxrishabh replied:
@Aryan-sharma11 I test all our go.mod files using snyk CLI. Other than `golang.org/x/net/http2@0.20.0`(low risk vuln, fix yet to implemented upstream) I can't find any vulnerability. Also, [github.com/go-jose/go-jose](https://security.snyk.io/vuln/SNYK-GOLANG-GITHUBCOMGOJOSEGOJOSE-6070736) is a transitive dependency to a package. Can we ignore this?

DelusionalOptimist committed:
chores and address review comments

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

DelusionalOptimist committed:
syncup go mods; lint & protos

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

daemon1024 committed:
fix(monitor): Missing Source in Telemetry

We did not have fallback logic to read from procfs in case of operation File/Network

Enabled the fallback logic for file/network telemetry

Signed-off-by: daemon1024 <barun1024@gmail.com>

rootxrishabh committed:
go.sum updated

Signed-off-by: rootxrishabh <risrock02@gmail.com>

daemon1024 committed:
Merge pull request #1448 from DelusionalOptimist/delusionaloptimist/feat/extended-vm-support

feat(*): extended VM support

DelusionalOptimist committed:
Merge pull request #1587 from daemon1024/fix-no-source-alerts

fix(monitor): Missing Source in Telemetry

rootxrishabh committed:
Conflicts resolved

Signed-off-by: rootxrishabh <risrock02@gmail.com>

rootxrishabh committed:
Merge branch 'updateDependencies' of https://github.com/rootxrishabh/KubeArmor into updateDependencies

Signed-off-by: rootxrishabh <risrock02@gmail.com>
